gnomad/utils/vcf.py

```diff
@@ -272,83 +277,99 @@
             logger.warning(
                 f"Coercing field info.{f} from int64 to int32 for VCF output. Value will be capped at int32 max value."
             )
-            info_ht = info_ht.annotate(
-                info=info_ht.info.annotate(
-                    **{f: hl.int32(hl.min(2 ** 31 - 1, info_ht.info[f]))}
-                )
-            )
+            info_expr.update({f: hl.int32(hl.min(2 ** 31 - 1, ht.info[f]))})
         elif ft == hl.dtype("array<int64>"):
             logger.warning(
                 f"Coercing field info.{f} from array<int64> to array<int32> for VCF output. Array values will be capped at int32 max value."
             )
-            info_ht = info_ht.annotate(
-                info=info_ht.info.annotate(
-                    **{
-                        f: info_ht.info[f].map(
-                            lambda x: hl.int32(hl.min(2 ** 31 - 1, x))
-                        )
-                    }
-                )
+            info_expr.update(
+                {f: ht.info[f].map(lambda x: hl.int32(hl.min(2 ** 31 - 1, x)))}
             )
 
+    ht = ht.annotate(info=ht.info.annotate(**info_expr))
+
     info_expr = {}
 
     # Make sure to pipe-delimit fields that need to.
     # Note: the expr needs to be prefixed by "|" because GATK expect one value for the ref (always empty)
-    # Note2: this doesn't produce the correct annotation for AS_SB_TABLE, but it is overwritten below
+    # Note2: this doesn't produce the correct annotation for AS_SB_TABLE, it is handled below
     for f in pipe_delimited_annotations:
-        if f in info_ht.info:
-            info_expr[f] = "|" + get_pipe_expr(info_ht.info[f])
+        if f in ht.info and f != "AS_SB_TABLE":
+            info_expr[f] = "|" + get_pipe_expr(ht.info[f])
 
     # Flatten SB if it is an array of arrays
-    if "SB" in info_ht.info and not isinstance(
-        info_ht.info.SB, hl.expr.ArrayNumericExpression
-    ):
-        info_expr["SB"] = info_ht.info.SB[0].extend(info_ht.info.SB[1])
+    if "SB" in ht.info and not isinstance(ht.info.SB, hl.expr.ArrayNumericExpression):
+        info_expr["SB"] = ht.info.SB[0].extend(ht.info.SB[1])
 
-    if "AS_SB_TABLE" in info_ht.info:
+    if "AS_SB_TABLE" in ht.info:
         info_expr["AS_SB_TABLE"] = get_pipe_expr(
-            info_ht.info.AS_SB_TABLE.map(lambda x: hl.delimit(x, ","))
+            ht.info.AS_SB_TABLE.map(lambda x: hl.delimit(x, ","))
         )
 
-    # Annotate with new expression and add 's' empty string field required to cast HT to MT
-    info_ht = info_ht.annotate(
-        info=info_ht.info.annotate(**info_expr), s=hl.null(hl.tstr)
-    )
+    # Annotate with new expression
+    ht = ht.annotate(info=ht.info.annotate(**info_expr))
+
+    return ht
+
 
-    # Create an MT with no cols so that we acn export to VCF
-    info_mt = info_ht.to_matrix_table_row_major(columns=["s"], entry_field_name="s")
-    return info_mt.filter_cols(False)
+def ht_to_vcf_mt(
```

Comment:
yeah I kept it because of the change to MT and the because it might already be used by others

Reply:
based on the hail docs, I'm not sure this transition to MT makes a difference in the export:
```
The FORMAT field is generated from the entry schema, which must be a tstruct. There is a FORMAT field for each field of the struct. If dataset is a Table, then there will be no FORMAT field and the output will be a sites-only VCF.
```
the datasets created from both `ht_to_vcf_mt` or `adjust_vcf_incompatible_types` will only be exported as sites-only vcfs, right?

Reply:
Yeah, thats true, do you think I should just remove this and be sure to point out in the change log that this removal is a breaking change? I didn't want to keep the name because we are no longer changing to a MT

Reply:
yeah, might be annoying to fix in pipelines but I think removing this makes sense since all of the functionality is included in the other function